pkg/log/handler_sentry.go

```diff
@@ -0,0 +1,79 @@
+package log
+
+import (
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/getsentry/sentry-go"
+	"time"
+)
+
+//go:generate mockery -name Sentry
+type Sentry interface {
+	CaptureException(exception error, hint *sentry.EventHint, scope sentry.EventModifier) *sentry.EventID
+}
+
+func init() {
+	AddHandlerFactory("metric", LoggerHandlerFactory)
+}
+
+func LoggerHandlerFactory(config cfg.Config, _ int) (Handler, error) {
+	return NewHandlerSentry(config), nil
+}
+
+type HandlerSentry struct {
+	sentry Sentry
+	tags   map[string]string
+	extra  map[string]interface{}
+}
+
+func NewHandlerSentry(config cfg.Config) *HandlerSentry {
+	env := config.GetString("env")
+	appName := config.GetString("app_name")
+
+	tags := map[string]string{
+		"application": appName,
+	}
+
+	client, _ := sentry.NewClient(sentry.ClientOptions{
+		Environment: env,
+	})
+
+	return &HandlerSentry{
+		sentry: client,
+		tags:   tags,
+		extra:  make(map[string]interface{}),
+	}
+}
+
+func (h HandlerSentry) WithExtra(extra map[string]interface{}) *HandlerSentry {
+	newExtra := mergeFields(h.extra, extra)
+
+	return &HandlerSentry{
+		sentry: h.sentry,
+		extra:  newExtra,
+	}
+}
+
+func (h HandlerSentry) Channels() []string {
+	return []string{}
+}
+
+func (h HandlerSentry) Level() int {
+	return PriorityError
+}
+
+func (h HandlerSentry) Log(_ time.Time, _ int, _ string, _ []interface{}, err error, data Data) error {
```

Comment:
Those are a lot of unused parameters; does it make sense to split the use cases of logging errors vs. logging the rest on handler level?

Reply:
I don't think so.. yes, the handler don't uses the parameter but i could imagine a handler which actually uses all of these..  
Isn't the io writer handler such a case? It uses all arguments on error..